### ISSUE TITLE
Maybe Cache Manager can be more flexible?

### ISSUE THREAD

**Issue by FastNinja** [renefloor/flutter_cache_manager#5]
Hi Rene,

```Information about the files is stored in the shared preferences with the key "lib_cached_image_data". (Because images was the first use of this library :)) The date when the cache is last cleaned is stored as "lib_cached_image_data_last_clean".```

You can actually pass this preferences in a Cache constructor (or in a `getInstance()` method) and your network image library will pass these keys (ib_cached_image_data).

Second point - it will probably be good to be able to have multiple cache managers. For example I will need to cache both document and data - and not sure if I will have same clean/keep configurations for both...

**Comment by FastNinja** [comment]
Another problem -like CacheManager generates unique ID when storing files locally.
on IOS the dialog that can preview documents requires a Path to the local copy and it reads and display it.
My app knows the actual file name but it is gone and Generated guid is presented  (see screenshot below)
![image](https://user-images.githubusercontent.com/13673826/36059185-f2d76c22-0e86-11e8-82ce-96522a5b03e7.png)

so basically I would like to be able to set a local file name for a file. I guess CacheManager need to created subfolder with GUID to make sure that file name are not clashed...

Any ideas?

**Comment by renefloor** [comment]
I already have been thinking about the possibility of multiple cache managers. Using a custom key for that is indeed a good idea. Keeping the filename is indeed also not a bad idea.

**Comment by Panthro** [comment]
This would definitely help with #20 !

**Comment by lifenautjoe** [comment]
Flexible could be applied also to being able to flush the cache programatically. When the user logs out of the application leaving dangling cache files with user photos or more is a privacy concern. I see theres a `_cleanCache` method but is private. Would be great to have it exposed in the public API.

**Comment by renefloor** [comment]
`0.3.0-alpha.2` adds a lot of flexibility. Love to hear more specific feature requests if this is not enough.
